projects/python-client/src/datapane/client/api/ipython_utils.py

```diff
@@ -155,11 +156,20 @@ def cells_to_blocks(ipython_output_cache: dict, opt_out: bool = True) -> typing.
 
     ..note:: IPython output caching must be enabled for this function to work. It is enabled by default.
     """
+    from IPython import get_ipython
+
     display_msg("Converting cells to blocks.")
     display_msg(
         "Please ensure all cells in the notebook have been executed and then saved before running this command."
     )
 
+    if not caller_name:
+        caller_name = cells_to_blocks.__name__
```

Comment:
can you provide some details / comment on what `caller_name` argument is used for?

Also, the following pattern is often used/useful in the codebase to populate optional fields

```python
caller_name = caller_name or cells_to_blocks.__name__
```

Reply:
I'll add a `# comment` for this. It's so the notebook converting function that's invoked by the user is known once we get to `cells_to_blocks`, so we can make sure its output, whatever it is, is not included.

I'll update to the pattern us throughout the codebase as recommended!

Reply:
Can we remove the caller requirement to provide `caller_name` and do via inspecting the call stack. Doesn't feel right to ask the caller to provide its identity.

Happy for us to spend more time getting the DX right here over getting out quickly.